apps/fraud/validation/api/src/main.py

```diff
@@ -0,0 +1,147 @@
+import logging
+import os
+import sys
+import time
+from datetime import timedelta
+
+import mlflow
+import mlflow.pyfunc
+from catboost import CatBoostClassifier
+from fastapi import Depends, FastAPI, HTTPException, Request, status
+from fastapi.logger import logger
+from fastapi.security import OAuth2PasswordRequestForm
+from fastapi_versioning import VersionedFastAPI, version
+from pcvalidation.core.extract_embedding import extract_embedding
+from pcvalidation.core.predict import get_prediction, get_prediction_main_contribution
+from pcvalidation.core.preprocess import convert_data_to_catboost_pool, preprocess
+from pcvalidation.utils.cloud_logging.logger import CustomLogger
+from pcvalidation.utils.cloud_logging.middleware import (
+    LoggingMiddleware,
+    cloud_trace_context,
+    http_request_context,
+)
+from pcvalidation.utils.cloud_logging.setup import setup_logging
+from pcvalidation.utils.configs import default_config as params
+from pcvalidation.utils.data_model import Item, Token, User, model_params
+from pcvalidation.utils.env_vars import (
+    ENV_SHORT_NAME,
+    LOGIN_TOKEN_EXPIRATION,
+    MLFLOW_CLIENT_ID,
+    users_db,
+)
+from pcvalidation.utils.security import (
+    authenticate_user,
+    create_access_token,
+    get_current_active_user,
+)
+from pcvalidation.utils.tools import connect_remote_mlflow
+from sentence_transformers import SentenceTransformer
+from starlette.requests import Request
+from typing_extensions import Annotated
+
+app = FastAPI(title="Passculture offer validation API")
+
+if ENV_SHORT_NAME == "dev":
+    setup_logging()
+    app.add_middleware(LoggingMiddleware)
+else:
+    logger = logging.getLogger("uvicorn")
+
+
+# Load latest model from MLFlow registery
+connect_remote_mlflow(MLFLOW_CLIENT_ID)
+model = CatBoostClassifier(one_hot_max_size=65)
+model_loaded = mlflow.catboost.load_model(
+    model_uri=f"models:/compliance_model_{ENV_SHORT_NAME}/Production"
+)
+
+# Encoding models
+TEXT_MODEL = SentenceTransformer("sentence-transformers/all-MiniLM-L6-v2")
+IMAGE_MODEL = SentenceTransformer("clip-ViT-B-32")
+
+
+@app.get("/")
+def read_root():
+    logger.info("Auth user welcome to : Validation API test")
+    return "Auth user welcome to : Validation API test"
+
+
+@app.post("/model/compliance/scoring")
+@version(1, 0)
+def get_item_validation_score(
+    request: Request,
+    item: Item,
+    current_user: Annotated[User, Depends(get_current_active_user)],
+):
+    start = time.time()
+    custom_logger = CustomLogger(request)
```

Comment:
Le logger devrait être initialisé via un import global
 
On peut rajouter un decorator aux fonctions http afin de mettre le `cloud_trace_context` de la request.
https://stackoverflow.com/questions/64497615/how-to-add-a-custom-decorator-to-a-fastapi-route